task-impls/src/network.rs

```diff
@@ -37,11 +35,10 @@ pub struct NetworkTaskState<
     MEMBERSHIP: Membership<TYPES>,
     COMMCHANNEL: CommunicationChannel<TYPES, Message<TYPES, I>, PROPOSAL, VOTE, MEMBERSHIP>,
 > {
-    channel: COMMCHANNEL,
-    events: UnboundedStream<SequencingHotShotEvent<TYPES, I>>,
-    event_stream: ChannelStream<SequencingHotShotEvent<TYPES, I>>,
-    view: ViewNumber,
-    phantom: PhantomData<(TYPES, Message<TYPES, I>, PROPOSAL, VOTE, MEMBERSHIP)>,
+    pub channel: COMMCHANNEL,
+    pub event_stream: ChannelStream<SequencingHotShotEvent<TYPES, I>>,
+    pub view: ViewNumber,
+    pub phantom: PhantomData<(TYPES, Message<TYPES, I>, PROPOSAL, VOTE, MEMBERSHIP)>,
```

Comment:
I wonder if this can be simplified slightly:
```suggestion
    pub phantom: PhantomData<(PROPOSAL, VOTE, MEMBERSHIP)>,
```